docs/prod-install.md

```diff
@@ -4,97 +4,190 @@ Installing Zulip in production
+`sudo -i` to start an interactive root shell.
+
+First, install the Let's Encrypt client [Certbot](https://certbot.eff.org/) and
+then generate the certificate:
+
+```
+apt-get install -y git bc openssl
+git clone https://github.com/certbot/certbot /opt/letsencrypt
+cd /opt/letsencrypt
+./certbot-auto certonly --standalone
+```
+
+Note: If you already had a webserver installed on this system (e.g. you
+previously installed Zulip and are now getting a cert), you will
+need to stop the webserver (e.g. `service nginx stop`) and start it
+again after (e.g. `service nginx start`) running the certbot command above.
+
+Next, symlink the certificates to make them available where Zulip expects them.
+Be sure to replace YOUR_DOMAIN with your domain name.
+
+```
+ln -s /etc/letsencrypt/live/YOUR_DOMAIN/privkey.pem /etc/ssl/private/zulip.key
+ln -s /etc/letsencrypt/live/YOUR_DOMAIN/fullchain.pem /etc/ssl/certs/zulip.combined-chain.crt
+```
+
+Note: Certificates provided by Let's Encrypt are valid for 90 days and then
+need to be [renewed](https://certbot.eff.org/docs/using.html#renewal). You can
+renew with this command:
+
+```
+./certbot-auto renew
+```
+
+### Using a self-signed certificate
+
+If you aren't able to use Let's Encrypt, you can generate a self-signed ssl certificate.
+
+Run all of these commands as root. If you're not already logged in as root, use
+`sudo -i` to start an interactive root shell.
+
+```
+apt-get install openssl
+openssl genrsa -des3 -passout pass:x -out server.pass.key 4096
+openssl rsa -passin pass:x -in server.pass.key -out zulip.key
+rm server.pass.key
+openssl req -new -key zulip.key -out server.csr
+openssl x509 -req -days 365 -in server.csr -signkey zulip.key -out zulip.combined-chain.crt
+rm server.csr
+cp zulip.key /etc/ssl/private/zulip.key
+cp zulip.combined-chain.crt /etc/ssl/certs/zulip.combined-chain.crt
+```
+
+You will eventually want to get a properly signed SSL certificate
+(and note that at present the Zulip desktop app doesn't support
+self-signed certificates), but this will let you finish the
+installation process. When you do get an actual certificate, you
+will need to install as /etc/ssl/certs/zulip.combined-chain.crt the
+full certificate authority chain, not just the certificate; see the
+section on "SSL certificate chains" [in the nginx
+docs](http://nginx.org/en/docs/http/configuring_https_servers.html)
+for how to do this.
+
+### If you are using a self-signed certificate with an IP address (no fqdn)
+
+Finally, if you want to proceed with just an IP address, it is
+possible to finish a Zulip installation that way; just set
+EXTERNAL_HOST to be the IP address.
+
+## Step 2: Download and uppack latest release
+
+Download [the latest built server
+tarball](https://www.zulip.com/dist/releases/zulip-server-latest.tar.gz) and
+unpack it to `/root/zulip`.
+
+Run all of these commands as root. If you're not already logged in as root, use
+`sudo -i` to start an interactive root shell.
```

Comment:
I added this because I always disable root login via ssh and so I login as my user and then sudo -i to drop in to root. Because we're not showing the commands with the command prompt including which user is logged in, I thought it would be good to be specific. I can also see how this would be extraneous info to an experienced sysadmin.